contracts/BC_fusion/AirDrop.sol

```diff
@@ -0,0 +1,128 @@
+pragma solidity 0.8.17;
+
+import "@openzeppelin/contracts-upgradeable/security/ReentrancyGuardUpgradeable.sol";
+import "@openzeppelin/contracts/utils/cryptography/MerkleProof.sol";
+import "@openzeppelin/contracts/utils/cryptography/ECDSA.sol";
+
+import "./interface/ITokenHub.sol";
+import "./interface/IAirDrop.sol";
+import "./System.sol";
+import "./lib/Utils.sol";
+
+contract AirDrop is IAirDrop, ReentrancyGuardUpgradeable, System {
+    string public constant sourceChainID = "Binance-Chain-Ganges";
+    address public approvalAddress = 0xaAaAaAaaAaAaAaaAaAAAAAAAAaaaAaAaAaaAaaAa;
+    bytes32 public merkleRoot = 0x0000000000000000000000000000000000000000000000000000000000000000;
+    bool public merkleRootAlreadyInit = false;
+
+    // claimedMap is used to record the claimed token.
+    mapping(bytes32 => bool) private claimedMap;
+
+    function isClaimed(bytes32 node) public view override returns (bool) {
+        return claimedMap[node];
+    }
+
+    function claim(
+        bytes32 tokenSymbol, uint256 amount,
+        bytes calldata ownerPubKey, bytes calldata ownerSignature, bytes calldata approvalSignature,
```

Comment:
only enable when `merkleRootAlreadyInit` is true.

Also please check length of `merkleProof` is not zero.

Reply:
fixed